provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -0,0 +1,357 @@
+    let currency_sign_index = pattern.find(currency_sign).unwrap();
+    let first_num_index = pattern.find(['0', '#']).unwrap();
+    let last_num_index = pattern.rfind(['0', '#']).unwrap();
+
+    let letters_set = match load_for_general_category_group(provider, GeneralCategoryGroup::Letter)
+    {
+        Ok(letters_set) => letters_set,
+        Err(icu_properties::PropertiesError::PropDataLoad(e)) => {
+            return Err(e);
+        }
+        Err(_) => unreachable!("load_for_general_category_group should only return PropDataLoad"),
+    };
+
+    let char_closer_to_number = {
+        if currency_sign_index < first_num_index {
+            place_holder.chars().next_back().unwrap()
+        } else if currency_sign_index > last_num_index {
+            place_holder.chars().next().unwrap()
+        } else {
+            return Err(DataError::custom(
+                "Currency sign must be in the middle of the pattern",
+            ));
+        }
+    };
+
+    if letters_set.as_borrowed().contains(char_closer_to_number) {
+        Ok(PatternSelection::StandardAlphaNextToNumber)
+    } else {
+        Ok(PatternSelection::Standard)
+    }
+}
+
+impl DataProvider<CurrencyEssentialsV1Marker> for crate::DatagenProvider {
+    fn load(
+        &self,
+        req: DataRequest,
+    ) -> Result<DataResponse<CurrencyEssentialsV1Marker>, DataError> {
+        self.check_req::<CurrencyEssentialsV1Marker>(req)?;
+        let langid = req.locale.get_langid();
+
+        let currencies_resource: &cldr_serde::currencies::Resource = self
+            .source
+            .cldr()?
+            .numbers()
+            .read_and_parse(&langid, "currencies.json")?;
+
+        let numbers_resource: &cldr_serde::numbers::Resource = self
+            .source
+            .cldr()?
+            .numbers()
+            .read_and_parse(&langid, "numbers.json")?;
+
+        let result = extract_currency_essentials(self, currencies_resource, numbers_resource);
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(result?)),
+        })
+    }
+}
+
+impl IterableDataProvider<CurrencyEssentialsV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(self
+            .source
+            .cldr()?
+            .numbers()
+            .list_langs()?
+            .map(DataLocale::from)
+            .collect())
+    }
+}
+
+fn extract_currency_essentials<'data>(
+    provider: &DatagenProvider,
+    currencies_resource: &cldr_serde::currencies::Resource,
+    numbers_resource: &cldr_serde::numbers::Resource,
+) -> Result<CurrencyEssentialsV1<'data>, DataError> {
+    let currencies = &currencies_resource.main.value.numbers.currencies;
+
+    let currency_formats = &&numbers_resource
+        .main
+        .value
+        .numbers
+        .numsys_data
+        .currency_patterns
+        .get(&tinystr!(8, "latn"))
```

Comment:
Issue: just noticed this. You should file a follow-up issue since these patterns might be numbering system dependent.

Reply:
done